### ISSUE TITLE
I have an idea

### ISSUE THREAD

**Issue by Plousios** [nitroshare/nitroshare-android#2]
Why not look into building an xposed module or TWRP flashable platform 1st for nitroshare on android? My thoughts are that could potentially bring in a lot more resources from xda/xposed developers and get a lot more people involved in the project. It also might be easier to integrate nitroshare into android from source code via xposed module or a twrp flashable platform, bypassing all the "red tape" so to speak with having the nitroshare platform being able to to be integrated into the andorid system with root capabilities. If there is anything I can do to help with this project, I want to help tell me what you need. If my idea seems silly and outlandish I apologize and thank you for putting up with me.

**Comment by nathan-osman** [comment]
Don't feel bad about the suggestion - I'm glad to see people interested!

I'm not quite sure I understand why an xposed module would be necessary though - it should be possible to design the app in such a way that it only requires standard APIs. The current method for discovery is IPv4 broadcast but this is being replaced with mDNS (IPv4/6 multicast) in the next release. Many other apps in the Play store are using mDNS without issues.

The Android app is on hold for the next few weeks while I work hard on finalizing the API for the next desktop version. I am hoping to have that complete by the end of the month. Once that is done, I will continue working on the next desktop release at the same time as the Android app. There will also likely be an Ubuntu Touch app since QML can take advantage of most of the Qt code as-is.

**Comment by eboye** [comment]
The best way I can think of for the app is this:

Files are shared via standard Android share option
Files are received in background to specific folder
The app itself could have the share popup for selecting discoverable devices and maybe the simple file explorer for exploring that folder for received files.

**Comment by nathan-osman** [comment]
Yes, that's pretty much exactly how I planned it to work.

**Comment by techtonik** [comment]
Having Android *nodes* with peer discovery and sharing shipped by default in firmware is awesome.
